### PR TITLE
Circleci/2.0

### LINKED ISSUE #201: Update to CircleCI 2.0
The project is configured using CircleCI 1.0 which will be [EOL](https://circleci.com/blog/sunsetting-1-0/) on 31/08.

[This](https://circleci.com/sunset1-0/) guide can be used to help migration.
Also, as indicated [here](https://circleci.com/docs/2.0/migrating-from-1-2/#using-the-10-to-20-config-translation-endpoint), there seems to an end point in the CircleCI API that performs an automatic migration (or at least jump starts the process). I would do that, but I don't rights to the project so I can't really use the endpoint

### PR DESCRIPTION
**Pull request by t-buss**
Close #201 

This _should_ be the right configuration, although I cannot confirm it (CI, ammirite?).
Really not sure about the `notify` block, as I couldn't find the page in the documentation.
Hope it works, nontheless.

### CODE REVIEW

**Comment by geoand** [comment]
The reason I didn't do this was that I didn't have access to the project and therefore couldn't use the automatic conversion tool. But FWIW it looks good to me :)

**Comment by tbuss93** [comment]
Pipeline fails with this error message:
`
[ERROR] Failures: 
[ERROR]   RibbonFallbackTest.testFallBackGreetingEndpoint:114 expected:<[No instances available for testapp]> but was:<[404 Client Error]>
`

Not sure if the CircleCI 2.0 pipeline caused this... It's using Docker containers instead of virtual machines like in CircleCI 1.0. Could that have altered the test environment?

**Comment by tbuss93** [comment]
Also, the dependencies step takes over a minute, while 1.0 builds did it in merely 16 seconds

**Comment by geoand** [comment]
* The ribbon test seems to be failing for various  PRs, it's probably a case of a flaky test
* With CircleCI 2 you need to explicitly handle the caching of the maven dependencies, something like [this](https://github.com/snowdrop/spring-boot-cache-booster/blob/master/.circleci/config.yml)

**Comment by tbuss93** [comment]
I added the caching directives to the yaml file and it indeed speeded up the build. Test is still failing though...

Since it seems to be happening to various PRs, I'll leave it up to the core maintainers to decide what to do about it.

**Comment by geoand** [comment]
Cool!

**Comment by geoand** [comment]
@t-buss would you like to rebase onto master? The flaky should be fixed there

**Comment by pivotal-issuemaster** [comment]
@t-buss Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=207)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=207) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@t-buss Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=207)!

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            ./mvnw -s .settings.xml -U --fail-never dependency:go-offline -Dservice.occurence=${_SERVICE_OCCURENCE} || true
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: spring-cloud-kubernetes-{{ .Branch }}-{{ checksum "pom.xml" }}
+      - run:
+          name: test
+          command: |
+            ./mvnw -s .settings.xml clean install -Dservice.occurence=${_SERVICE_OCCURENCE} #org.jacoco:jacoco-maven-plugin:prepare-agent install -U -P sonar -nsu --batch-mode -Dmaven.test.redirectTestOutputToFile=true -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+            find . -type f -regex ".*/spring-cloud-*.*/target/*.*" | cpio -pdm $CIRCLE_ARTIFACTS
```
Comment (t-buss) [inline .circleci/config.yml]: The Docker images does not include the `cpio` command. Can't we just use `cp` to copy the artifacts?

Comment (geoand) [reply .circleci/config.yml]: I don't see why not (although I'm not familiar with the CircleCI setup of the project)...

**Comment by t-buss** [comment]
Boy, I tell ya': Those little mistakes you don't find locally 😅 
Anyway, pipeline seems to be working now. 🎉

**Comment by geoand** [comment]
Yeah, not being able to test things locally is a real drag :P.

Great work! @iocanel Can we merge this? The existing CircleCI configuration will be discontinued in a little over of a month.

**Pull request merged**